### PR TITLE
controller pattern

### PR DESCRIPTION
**Pull request by charlierudolph**
@alexdavid @alexpeachey @kevgo 

join `controllersDirPath` and `controllersWhitelist` together into one by specifying a glob pattern.

### CODE REVIEW

File: test/feature/finding_controllers.feature [hunk 1]
```diff
@@ -48,30 +48,7 @@ Feature: Finding controllers
           res.end 'user list'
       module.exports = UsersController
       """
-    And an exprestive app with the option "controllersDirPath" set to `"./my_controllers"`
-    When making a GET request to "/users"
-    Then the response code should be 200
-    And the response body should be "user list"
-
-
-  Scenario: an application with a restrictive controller whitelist
```
Comment (charlierudolph) [inline test/feature/finding_controllers.feature:57]: This test seemed unnecessary to me and is hard to recreate with glob.

File: test/feature/finding_controllers.feature [hunk 2]
```diff
@@ -48,30 +48,7 @@ Feature: Finding controllers
...
-    Given a file "controllers/not_a_controller.coffee" with the content
-      """
-      throw 'not a controller'
-      """
-    And an exprestive app with the option "controllersWhitelist" set to `/^(?!not_a_controller).coffee$/`
-    Then the app doesn't error
-
-
-  Scenario: an application with a different controller file name format
```
Comment (charlierudolph) [inline test/feature/finding_controllers.feature:66]: I think we only need one test for the custom pattern so removed this test as well

File: test/feature/finding_controllers.feature (hunk 1 above)
Comment (alexdavid) [inline test/feature/finding_controllers.feature:57]: Yeah, now that we're using glob for both `controllersDirPath` and `controllersWhitelist` this test isn't necessary anymore.  It was mainly just testing  `controllersWhitelist` functionality

File: src/exprestive.coffee [hunk 3]
```diff
@@ -11,8 +11,7 @@ class Exprestive
   @defaultOptions =
     appDir: ''
     routesFilePath: 'routes'
-    controllersDirPath: 'controllers'
-    controllersWhitelist: /^.+_controller\.(?:coffee|js)$/
+    controllersPattern: 'controllers/*_controller.{coffee,js}'
```
Comment (kevgo) [inline src/exprestive.coffee:14]: Good move!  +1 on this change!

File: test/spec/exprestive_spec.coffee [hunk 4]
```diff
@@ -22,7 +22,11 @@ describe 'Exprestive configuration', ->
       expect(@RoutesInitializerMock).to.have.been.calledWith '/base/dir/routes'
 
     it 'uses the default controller directory', ->
-      expect(@ControllerInitializerMock).to.have.been.calledWith '/base/dir/controllers'
+      expectedArg = sinon.match
+        appDir: '/base/dir'
+        controllersPattern: 'controllers/*_controller.{coffee,js}'
+        dependencies: {}
+      expect(@ControllerInitializerMock).to.have.been.calledWith expectedArg
```
Comment (kevgo) [inline test/spec/exprestive_spec.coffee]: You could inline the variable `expectedArg` while retaining readability and understandability:

``` coffeescript
expect(@ControllerInitializerMock).to.have.been.calledWith sinon.match
  appDir: '/base/dir'
  controllersPattern: 'controllers/*_controller.{coffee,js}'
  dependencies: {}
```

**Comment by kevgo** [comment]
:ship: from my end

**Comment by alexdavid** [comment]
Looks good.  :ship: from me too

File: test/spec/exprestive_spec.coffee (hunk 4 above)
Comment (charlierudolph) [inline test/spec/exprestive_spec.coffee]: changed

**Pull request closed without merging**